networkit/cpp/centrality/ComplexPaths.cpp

```diff
@@ -0,0 +1,205 @@
+/*
+ *  ComplexPaths.cpp
+ *
+ *
+ *  Created on:16.06.2023
+ *      Author: Klaus Ahrens
+ *              <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented from
+ *
+ *            https://github.com/drguilbe/complexpaths.git
+ *
+ *  see [ Guilbeault, D., Centola, D. Topological measures for
+ *        identifying and predicting the spread of complex contagions.
+ *        Nat Commun 12, 4430 (2021).
+ *        https://doi.org/10.1038/s41467-021-24704-6 ]
+ *
+ */
+
+#include <algorithm>
+#include <iostream>
+#include <iterator>
+#include <random>
+#include <vector>
+
+#include <networkit/auxiliary/Random.hpp>
+#include <networkit/base/Algorithm.hpp>
+#include <networkit/centrality/ComplexPaths.hpp>
+#include <networkit/distance/BFS.hpp>
+#include <networkit/distance/PrunedLandmarkLabeling.hpp>
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+
+ComplexPathAlgorithm::ComplexPathAlgorithm(const Graph &G, count threshold, Mode mode, node start)
+    : inputGraph{&G}, mode{mode}, start{start}, threshold{threshold}, normPaths{false} {
```

Comment:
Add a check for `threshold > 0`. Below in the code you have `auto seedsNeeded = threshold - 1` which will overflow if `threshold` is 0.